.github/workflows/check-dist.yml

```diff
@@ -0,0 +1,70 @@
+# In TypeScript actions, `dist/` is a special directory. When you reference
+# an action with the `uses:` property, `dist/index.js` is the code that will be
+# run. For this project, the `dist/index.js` file is transpiled from other
+# source files. This workflow ensures the `dist/` directory contains the
+# expected transpiled code.
+#
+# If this workflow is run from a feature branch, it will act as an additional CI
+# check and fail if the checked-in `dist/` directory does not match what is
+# expected from the build.
+name: Check Transpiled JavaScript
+
+on:
+  pull_request:
+    branches:
+      - '**'
+  push:
+    branches:
+      - main
+
+concurrency:
+  group: check-dist-${{ github.sha }}
+  cancel-in-progress: true
+
+permissions:
+  contents: read
+
+jobs:
+  check-dist:
+    name: Check dist/
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout
+        id: checkout
+        uses: actions/checkout@v4
+
+      - name: Setup Node.js
+        id: setup-node
+        uses: actions/setup-node@v4
+        with:
+          node-version-file: .node-version
+          cache: npm
+
+      - name: Install Dependencies
+        id: install
+        run: npm ci
+
+      - name: Build dist/ Directory
+        id: build
+        run: npm run build
+
+      # This will fail the workflow if the PR wasn't created by Dependabot.
```

Comment:
Why should the workflow fail if the PR wasn't created by Dependabot?

Reply:
I'm just as confused as you to be honest 🤷‍♂️. I took the action directly from the [TypeScript Template](https://github.com/actions/typescript-action/blob/1da1fe9abb959743f36210ae5423e1dccc805e85/.github/workflows/check-dist.yml), however nothing in the template or workflow backs up the comment. Perhaps there's some implicit Dependabot behaviour that we're not aware of. Either way, it's confusing probably not worth looking into since the workflow itself works perfectly. I'll remove this comment and the one below.